### ISSUE TITLE
Akka grpc maven don't have an option to add extra generators as like sbt

### ISSUE THREAD

**Issue by venkatt007** [akka/akka-grpc#1103]
### Versions used 
1.0.1

Akka version: 

### Expected Behavior
Should be able to add play java code generator in extra generators settings like in any

Please describe the expected behavior of the issue, starting from the first action.

### Actual Behavior
No option to add extra generators in maven pom file

Please provide a description of what actually happens, working from the same starting point.

### Relevant logs

If you identified a section in your logs that explains the bug or might be important to understand it, please add it.

### Reproducible Test Case

Please provide a PR with a failing test.

If the issue is more complex or requires configuration, please provide a link to a project that reproduces the issue.

**Comment by raboof** [comment]
In fact this option does exist, as an 'extraGenerators' option that takes a list of FQDN's: https://github.com/akka/akka-grpc/blob/master/maven-plugin/src/main/maven/plugin.xml#L52

I guess this feature needs to be better documented? Or did you find it didn't work for you?

**Comment by venkateshzifo** [comment]
Yeah its missing in documentation. I am looking to add akkaGrpcExtraGenerators += akka.grpc.gen.javadsl.play.PlayJavaServerCodeGenerator like we add in build.sbt to use gRPC endpoint

**Comment by venkateshzifo** [comment]
Also outputDirectory  path is configured but when i run maven compile it generates the file inside same target/generated-sources always

**Comment by raboof** [comment]
> outputDirectory path is configured but when i run maven compile it generates the file inside same target/generated-sources always

That's weird, I don't see anything obviously wrong in that part of the code. Could you file that as a separate issue?

**Comment by venkateshzifo** [comment]
Sure. Any idea on adding extra generators properly ? Is it configured properly ?

**Comment by raboof** [comment]
> Any idea on adding extra generators properly ? Is it configured properly ?

I'm not sure how to encode a list in a Maven configuration from the top of my head, but looking at https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Mapping_Lists it looks like you might have to do something like `<extraGenerators><extraGenerator>play.grpc.gen.javadsl.PlayJavaServerCodeGenerator</extraGenerator></extraGenerators>`

**Comment by venkateshzifo** [comment]
Tried the same as well. But still throws java.lang.ClassNotFoundException: play.grpc.gen.javadsl.PlayJavaServerCodeGenerator

**Comment by raboof** [comment]
> java.lang.ClassNotFoundException: play.grpc.gen.javadsl.PlayJavaServerCodeGenerator

This suggests the configuration is parsed correctly, but the play-grpc codegen classes are not on the maven builds' classpath. You probably need to add the play-grpc codegen jar as a dependency of the plugin something like https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Using_the_dependencies_Tag

**Comment by venkateshzifo** [comment]
> > java.lang.ClassNotFoundException: play.grpc.gen.javadsl.PlayJavaServerCodeGenerator
> 
> This suggests the configuration is parsed correctly, but the play-grpc codegen classes are not on the maven plugins' classpath. You probably need to add the play-grpc codegen jar as a dependency of the plugin something like https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Using_the_dependencies_Tag

Cool. It worked i added the dependency under the plugin

**Comment by raboof** [comment]
Nice! Let's leave this issue open to remind ourselves to add this to the documentation. Would you perhaps like to give that a shot?

**Comment by venkateshzifo** [comment]
Sure. I am planning to add this sample in lagom samples once i am able to run fully

**Comment by raboof** [comment]
Great, looking forward to it!

**Comment by venkateshzifo** [comment]
@raboof  Facing this issue now when starting lagom with mvn lagom:runAll Cannot load module[Module [example.myapp.helloworld.grpc.AkkaGrpcClientModule] cannot be instantiated.] facing this issue. Any idea on this ?

**Comment by venkateshzifo** [comment]
I am facing below issue in lagom gRPC with maven. GreeterServiceClientProvider.java:[32,32] method create in class com.zifo.cuttle.importservice.impl.grpc.GreeterServiceClient cannot be applied to given types;
[ERROR] required: akka.grpc.GrpcClientSettings,akka.actor.ClassicActorSystemProvider
[ERROR] found: akka.grpc.GrpcClientSettings,akka.stream.Materializer,scala.concurrent.ExecutionContextExecutor
[ERROR] reason: actual and formal argument lists differ in length

I understand there is version mismatch not sure which version of play code generator to use. Server stubs are creating fine only client stubs causing issues

**Comment by ignasi35** [comment]
@venkateshzifo I think this discussion is at a point where it'd be better to move it to a PR on https://github.com/lagom/lagom-samples/. I suggest you created a `DRAFT` PR so it'll be easier to investigate your code and reproduce the errors as you face them.

Since this has diverged from the original issue description I think we may close the issue too (I'll leave that to your discretion :-) ).
